src/SnackBar/SnackBar.js

```diff
@@ -0,0 +1,111 @@
+import {safeText} from "../utils.js";
+
+export default class SnackBar {
+    /** @param {string} - needed in order to be able to create multiple snackbars on a page */
+    #snackBarId;
+    /** @param {number} - needed in order to delete the previous timeout when a new one appears */
+    #timeoutId;
+    /** @param {HTMLElement} */
+    #snackBarEl;
+
+    /** @param {string} id */
+    constructor(id = 'snackBar') {
+        this.#snackBarId = id;
+    }
+
+    /**
+     * SnackBar is a tiny notification displayed at the bottom of the screen. It's allowed to have only one SnackBar in
+     * the same time. Creating a new SnackBar will remove the old one. Default living time is 15 sec after which
+     * SnackBar disappears.
+     * Properties:
+     *  - msgText - main text (e.g. Smth was deleted)
+     *  - btnText - OK button text (e.g. Undo)
+     *  - btnCb - function to be called when OK btn pressed
+     *  - ttl - aka TimeToLive in milliseconds
+     * @param {{msgText:string, btnText:string, btnCb:function, ttl:number}} prop
+     */
+    show(prop) {
+        SnackBar.#validateProperties(prop);
+        this.#removeExistingIfNeeded();
+        this.#snackBarEl = this.#createSnackBarElement(this.#snackBarId, prop);
+        this.#addEventListeners(prop.btnCb);
+        this.#timeoutId = setTimeout(() => {
+            this.#removeEl()
+        }, prop.ttl || 15 * 1000);
+    }
+
+    #removeExistingIfNeeded(){
+        const existing = document.getElementById(`${this.#snackBarId}`);
+        if (existing)
+            existing.remove();
+        if (this.#timeoutId)
+            window.clearTimeout(this.#timeoutId)
+    }
+
+    /** @param {Function} callback */
+    #addEventListeners(callback) {
+        const okButton = this.#snackBarEl.querySelector('[js-ok]');
```

Comment:
I'd move all the querySelector() code into separate methods so that it's not mixed with the main logic.